mlflow/utils/environment.py

```diff
@@ -531,6 +531,11 @@ def _generate_mlflow_version_pinning() -> str:
     # We reach here when mlflow is installed from the source outside of the MLflow CI environment
     # (e.g., Databricks notebook).
 
+    from mlflow.sha import SHA
+
+    if SHA:
```

Comment:
When would SHA be None?

Reply:
It's set when building a wheel. For example, when you install mlflow from your PR branch, pip builds a wheel first, then install it. When you use this mlflow, `SHA` is not None. Otherwise, it's None.